### ISSUE TITLE
can't apply conditions in one to one association

### ISSUE THREAD

**Issue by elrasco** [balderdashy/waterline#1238]
the method .populate( foreignKey, [query] ) doesn't apply the query in one to one association.
for example in a relation like this, where a video belong to a project, I can't get active project videos.
video.js
```javascript
module.exports = {

  attributes: {
  	title: 'string',
  	project_id: {
  		model: 'project'
  	}
  }
};
```
project.js
```javascript
module.exports = {
  attributes: {
  	status: 'string',	
  	title: 'string'
  }
};
```

The call
```javascript
Content
    .find()
    .populate('project_id', {'status': 'delivered'} ).exec(function(e, r) {
        ...
     })
    
```
doesn't filter the project_id association.
If waterline doesn't fully support one to one association, please document it.

**Comment by particlebanana** [comment]
You can get to this by going from the opposite side:

```javascript
// Video.js
module.exports = {
  attributes: {
    title: 'string',
    project_id: {
        model: 'project'
    }
  }
};
```

```javascript
// Project.js
module.exports = {
  attributes: {
    status: 'string',   
    title: 'string',
    videos: {
      collection: 'video',
      via: 'project_id'
    }
  }
};
```

```javascript
Project.find()
.where({ status: 'delivered' })
.populate('videos')
.exec(....)
```

Populates only "inflate" defined attributes.

**Comment by elrasco** [comment]
Thank you particlebanana... I knew that workaround but I can't use it. 
Actually I need to query also videos so the problem persits. 
Anyway, what I highlighted is that populate doesn't work as documented, I didn't need help.

**Comment by coagmano** [comment]
Thanks for pointing that out, I've made a pull request to make that explicit in the Sails docs.

**Comment by sailsbot** [comment]
Thanks for posting, @elrasco.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
